pkgs/by-name/pv/pvs-studio/package.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchurl
+
+, makeWrapper
+, perl
+, strace
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pvs-studio";
+  version = "7.30.80678.389";
+
+  src =
+    let
+      system = stdenv.hostPlatform.system;
+      selectSystem = attrs:
+        attrs.${system} or (throw "Unsupported system: ${system}");
+    in
+    fetchurl {
+      url = selectSystem {
+        x86_64-darwin = "https://cdn.pvs-studio.com/pvs-studio-${version}-macos.tgz";
+        x86_64-linux = "https://cdn.pvs-studio.com/pvs-studio-${version}-x86_64.tgz";
```

Comment:
No longer available for that version.

Reply:
urls going 404 is a problem, especially for unfree. Consider using archive.org

Reply:
Archiving both via [archive.org](https://web.archive.org) and [archive.is](https://archive.is) didn't work.

Reply:
Okay, I just noticed the archives actually appeared on the Wayback Machine after a while.